app/javascript/tapp/components/courseForm.js

```diff
@@ -184,48 +198,97 @@ class CourseForm extends React.Component {
-                                        if (event.target.value != this.props.course.qual) {
-                                            this.props.updateCourse(
-                                                this.props.courseId,
-                                                event.target.value,
-                                                'qual'
-                                            );
-                                        }
-                                    }}
-                                    defaultValue={this.props.course.qual}
-                                />
-                            </td>
-                            <td className="col-9">
-                                <p>
-                                    <b>Responsibilities: </b>
-                                </p>
-                                <textarea
-                                    onBlur={event => {
-                                        if (event.target.value != this.props.course.resp) {
-                                            this.props.updateCourse(
-                                                this.props.courseId,
-                                                event.target.value,
-                                                'resp'
-                                            );
-                                        }
-                                    }}
-                                    defaultValue={this.props.course.resp}
-                                />
-                            </td>
                         </tr>
                     </tbody>
                 </table>
+                <ExpandableDescriptions expanded={this.state.expanded}
+                                        handleClick={() => this.toggleExpanded()}
+                                        {...this.props}
+                />
             </ListGroupItem>
         );
     }
 }
 
+
+class ExpandableDescriptions extends React.Component {
+    render () {
+        if (this.props.expanded) {
+            return (
+                <div>
+                    <Descriptions {...this.props} key={this.props.courseId}/>
+                    <div>
+                        <button className="expand-button" onClick={() => this.props.handleClick()}>
+                            <span className="glyphicon glyphicon-chevron-up">
+                            </span>
+                        </button>
+                    </div>
+                </div>
+            );
+        } else {
+            return (
+                <div>
+                    <button className="expand-button" onClick={() => this.props.handleClick()}>
+                        <span className="glyphicon glyphicon-chevron-down">
+                        </span>
+                    </button>
+                </div>
+            );
+        }
+    }
+}
+
+class Descriptions extends React.Component {
+    render() {
+        return (
+            <div>
```

Comment:
Not sure, do we need to wrap in <div>, here and in other similar outer locations ?